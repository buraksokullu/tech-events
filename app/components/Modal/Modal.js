import React from 'react';
import PropTypes from 'prop-types';

import Button from 'Components/FormFields/Button/Button';
import s from './Modal.scss';

const Modal = ({ handleClose, show, children }) => {
  return (
    <div className={` ${s.modal} ${show ? s.displayBlock : s.displayNone}`}>
      <section className={s.modalMain}>
        {children}

        <div className={s.cancelBtn}>
          <Button onClick={handleClose}>Cancel</Button>
        </div>
      </section>
    </div>
  );
};

Modal.propTypes = {
  handleClose: PropTypes.func.isRequired,
  show: PropTypes.bool.isRequired,
  children: PropTypes.node.isRequired
};

export default Modal;
