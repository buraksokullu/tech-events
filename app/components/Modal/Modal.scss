.modal {
  display: none;
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background: rgba(0, 0, 0, 0.6);

  &.displayBlock {
    display: block;
  }

  &.displayNone {
    display: none;
  }

  .modalMain {
    position: fixed;
    background: white;
    font-size:16px;
    width: 100%;
    max-width: 600px;
    padding: 30px;
    height: auto;
    border-radius: 8px;;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);

    .cancelBtn{
      display: inline-block;
      margin-left: 20px;;
    }
    span{
        display: block;
        margin-bottom: 20px;
        color: $grey120;
        strong{
            margin: 0 5px;
        }
    }
  }
}
