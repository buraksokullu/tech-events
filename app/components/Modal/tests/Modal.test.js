import React from 'react';
import { shallow } from 'enzyme';
import Modal from '../Modal';

describe('Modal', () => {
  let modalProps = null;
  const handleCloseMock = jest.fn();
  beforeEach(() => {
    modalProps = {
      handleClose: handleCloseMock,
      show: false,
      children: 'children'
    };
  });

  it('Should render correctly', () => {
    const component = shallow(<Modal {...modalProps} />);

    expect(component).toMatchSnapshot();
  });
});
