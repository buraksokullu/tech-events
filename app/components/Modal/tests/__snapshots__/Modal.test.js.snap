// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Should render correctly 1`] = `
<div
  className=" undefined undefined"
>
  <section>
    children
    <div>
      <Button
        onClick={[MockFunction]}
      >
        Cancel
      </Button>
    </div>
  </section>
</div>
`;
